### PR TITLE
Migrate Citadel ca package metrics from prometheus to open census

### PR DESCRIPTION
**Pull request by Monkeyanator**
This PR migrates the metrics in the Citadel ca package from prometheus to open census. [Related issue](https://github.com/istio/istio/issues/14362) which calls for open census metrics in Citadel agent.

Parallel PR to [this](https://github.com/istio/istio/pull/15223) and [this](https://github.com/istio/istio/pull/15413)

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by Monkeyanator** [comment]
fyi @pitlv2109

File: tests/e2e/tests/dashboard/dashboard_test.go [hunk 1]
```diff
@@ -384,6 +384,19 @@ func citadelQueryFilterFn(queries []string) []string {
 		if strings.Contains(query, "secret_deleted_cert_count") {
 			continue
 		}
+		if strings.Contains(query, "server_csr_count") {
```
Comment (howardjohn) [inline tests/e2e/tests/dashboard/dashboard_test.go]: if you didn't add any new metrics why did you have to add more here?

Comment (Monkeyanator) [reply tests/e2e/tests/dashboard/dashboard_test.go]: I'm looking into that now. I believe this PR was added before the `dashboard_test` was merged in, so before I changed it today, all the tests were passing.

Comment (Monkeyanator) [reply tests/e2e/tests/dashboard/dashboard_test.go]: Confirmed that these metrics are exported / can be triggered and read, but have no data points ingested in this e2e test. Turns out due to prefixing issues, this test should actually fix a few of the dashboard panels (adding `citadel_server` prefix back as the dashboard expects).

At the moment, unfortunately, so many of the Citadel dashboard metrics are ignored that the test isn't covering much of anything save for the process metrics.

Comment (Monkeyanator) [reply tests/e2e/tests/dashboard/dashboard_test.go]: Found the difference. When we use a `promtheus.NewCounterVec`, it seems to initially use `0` as the count for the metric, whereas for `monitoring.NewSum`, it shows up as `no data`. That's why these metrics got past the tests initially.

**Comment by Monkeyanator** [comment]
I noticed that when migrating from prom metrics to the new metrics package, metrics which used to default to `0` when not present are now not defined until incremented the first time ([related to this most likely](https://github.com/census-instrumentation/opencensus-go/issues/935)). Therefore, as we continue migration, we'll have to ignore more metrics in the dashboard tests which cannot be invoked.

@douglas-reid any thoughts on this?

**Pull request merged**